src/Compiler/Scheme/Chez.idr

```diff
@@ -282,19 +312,25 @@ schemeCall fc sfn argns ret
 -- function call.
 useCC : {auto c : Ref Ctxt Defs} ->
         {auto l : Ref Loaded (List String)} ->
-        String -> FC -> List String -> List (Name, CFType) -> CFType -> Core (String, String)
-useCC appdir fc ccs args ret
-    = case parseCC ["scheme,chez", "scheme", "C"] ccs of
-           Nothing => throw (NoForeignCC fc)
+        String -> FC -> List String -> List (Name, CFType) -> CFType ->
+        Maybe Version -> Core (String, String)
+useCC appdir fc ccs args ret version
+    = case parseCC ["scheme,chez", "scheme", "C__collect_safe", "C"] ccs of
            Just ("scheme,chez", [sfn]) =>
                do body <- schemeCall fc sfn (map fst args) ret
                   pure ("", body)
            Just ("scheme", [sfn]) =>
                do body <- schemeCall fc sfn (map fst args) ret
                   pure ("", body)
-           Just ("C", [cfn, clib]) => cCall appdir fc cfn clib args ret
-           Just ("C", [cfn, clib, chdr]) => cCall appdir fc cfn clib args ret
-           _ => throw (NoForeignCC fc)
+           Just ("C__collect_safe", (cfn :: clib :: _)) => do
+             when (unsupportedCallingConvention version) $
+               throw (GenericMsg fc $ "Chez version " ++ show version ++  " is undefined or does not support '__collect_safe' calling convention")
+             cCall appdir fc cfn clib args ret True
```

Comment:
Wasn't the plan to keep generating code without the collect safe calling convention if
the available scheme wasn't recent enough?

Reply:
I think I could create an extra commit on top of Russoul's and Abailly's work to change the detection from an error if unsupported to switch to the safe on collect if supported.

Reply:
Changed to:

```
             if unsupportedCallingConvention version
               then cCall appdir fc cfn clib args ret False
               else cCall appdir fc cfn clib args ret True
```

Reply:
Ended up amending rather than creating an extra commit.